### ISSUE TITLE
[feat] Ability to broadcast to a set of nodes

### ISSUE THREAD

**Issue by LordMike** [OpenZWave/Zwave2Mqtt#750]
**Is your feature request related to a problem? Please describe.**

I'm trying to optimize a little, so that I can turn off 10+ lights in rapid succession. Right now, when I use HASS to turn off all lights, Z2M will get 10+ OFF commands via MQTT topics. In practice, I then see lights go off.. one by one.. sometimes with great delay between them.

I think that, by sending fewer messages on z-wave, I'll be better off. If I understand it correctly, broadcasting would be the way to go (hopefully this is actually fewer messages, and not just Z2M sending many messages ..).

**Describe the solution you'd like**
The current broadcast API seems to send messages to "all" devices (that support the command class). Would it be possible to target some devices, f.ex. by a regex on their node name, by listing node id's, or something else? 

**Describe alternatives you've considered**

In principle, I could use the associations of z-wave, and f.ex. dedicate an endpoint to the cause. Then w/ Z2M turn that endpoint on/off, and then have it send Basic Set's to all its associated nodes.. I could distribute the load by doing this a few times ... but that seems way too complicated.. 

**Additional context**
N/A

**Comment by robertsLando** [comment]
> If I understand it correctly, broadcasting would be the way to go

What do you mean? There are already broadcast apis in z2m, but the way them works are almost the same, once I receive a broadcast set I then make N zwave commands to specific nodes matching the broadcast: https://github.com/OpenZWave/Zwave2Mqtt#broadcast

**Comment by LordMike** [comment]
Damn - I had hoped Z-wave had a broadcast message.. or maybe a message with many "To" fields..

Oh well.. Would it be possible to expand this API, such that I could do a comma-separated list of nodes (names) to target - or perhaps a regex? 

Broadcasting a "basic set OFF", as I understand it, would turn off everything I have... lights, buttons (set their states to off), plugs, fans, .. .. right? So being able to target just a subset would be beneficial.

One way would be to, in the payload that is sent, have two well-known fields (assume it's json being sent):

```json
{
  .. other data ..
  "nodes": [ 3, 4, "wallswitch_1", "fan_ceiling" ],
  "nodes_regex": "wallswitch_[0-9]+"
}
```

If any of the two is set, it would limit the target of the "broadcast".

Note that my goal is to reduce latency .. it's entirely possible I can't save any time doing it this way, as sending N messages via MQTT myself doesn't take a lot of time .. .. I'll try the broadcast.. see if it turns off everything faster than right now.

**Comment by LordMike** [comment]
I've tested a little.. for future reference, I used the following topic:

> zwave2mqtt/_CLIENTS/ZWAVE_GATEWAY-(name)/broadcast/37/6/0/set

It's a BASIC_SET command class (37), and I was targeting endpoint 6 on my switches. They payload was "0".

I did not see any difference in time, when doing a HASS "off", where I click a toggle that sends a "turn off" command to all lights as quickly as possible.. and then doing this broadcast thing.

It seems to me, and I only timed this mentally, that both methods had variances in how fast they were.. I _assume_, that this is due to messages being lost, as I experience that when its "slow", it's always "10 seconds slower"... like a timeout of 10s was applied somewhere.

I don't think the Feature request is valid then - for my use case. I'll close it.
